src/game/ZoneArea.cxx

```diff
@@ -0,0 +1,153 @@
+#include "ZoneArea.hxx"
+#include "../engine/basics/PointFunctions.hxx"
+#include "../services/Randomizer.hxx"
+#include "../engine/TileManager.hxx"
+#include "../engine/Engine.hxx"
+
+void mergeZoneAreas(ZoneArea &mainZone, ZoneArea &toBeMerged)
+{
+  mainZone.m_zoneNodes.insert(mainZone.m_zoneNodes.end(), toBeMerged.m_zoneNodes.begin(), toBeMerged.m_zoneNodes.end());
+  mainZone.m_hasPower |= toBeMerged.m_hasPower;
+  mainZone.m_hasWater |= toBeMerged.m_hasWater;
+  mainZone.xmin = std::min(mainZone.xmin, toBeMerged.xmin);
+  mainZone.xmax = std::max(mainZone.xmax, toBeMerged.xmax);
+  mainZone.ymin = std::min(mainZone.ymin, toBeMerged.ymin);
+  mainZone.ymax = std::max(mainZone.ymax, toBeMerged.ymax);
+};
+
+ZoneArea::ZoneArea(ZoneNode zoneNode)
+    : m_zoneNodes{zoneNode}, xmin(std::max(0, zoneNode.coordinate.x - 1)), m_zoneType(zoneNode.zoneType),
+      m_zoneDensity(zoneNode.zoneDensity), xmax(std::min(Settings::instance().mapSize, zoneNode.coordinate.x + 1)),
+      ymin(std::max(0, zoneNode.coordinate.y - 1)), ymax(std::min(Settings::instance().mapSize, zoneNode.coordinate.y + 1))
+{
+}
+
+void ZoneArea::spawnBuildings()
+{
+
+  constexpr int amountOfBuildingsToSpawn = 5;
+  auto &randomizer = Randomizer::instance();
+  // shuffle mapNodes so placement of building looks random
+  randomizer.shuffle(m_zoneNodes.begin(), m_zoneNodes.end());
+
+  int buildingsSpawned = 0;
+
+  // pick every single zone node we have
+  for (auto &node : m_zoneNodes)
+  {
+    if (node.occupied)
+    {
+      continue;
+    }
+
+    if (buildingsSpawned >= amountOfBuildingsToSpawn)
+    {
+      break;
+    }
+    // get the maximum size we can spawn at this node
+    TileSize maxTileSize = {getMaximumTileSize(node.coordinate).width, getMaximumTileSize(node.coordinate).height};
+    std::string buildingTileID = TileManager::instance().getRandomTileIDForZoneWithRandomSize(getZone(), getZoneDensity(), maxTileSize);
```

Comment:
I would prefer direct access to the member attributes, instead of using functions.

Reply:
done